src/Microsoft.AspNet.Mvc.ModelBinding/ModelStateDictionary.cs

```diff
@@ -265,6 +268,24 @@ public ModelValidationState GetFieldValidationState([NotNull] string key)
         }
 
         /// <summary>
+        /// Returns <see cref="ModelValidationState"/> for the <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The key to look up model state errors for.</param>
+        /// <returns>Returns <see cref="ModelValidationState.Unvalidated"/> if no entry is found for the specified
+        /// key, <see cref="ModelValidationState.Invalid"/> if an instance is found with one or more model
+        /// state errors; <see cref="ModelValidationState.Valid"/> otherwise.</returns>
+        public ModelValidationState GetValidationState([NotNull] string key)
+        {
+            ModelState exactMatchValue;
```

Comment:
It's just a bit confusing having 2 different APIs that aren't well differentiated - `GetValidationState`, `GetFieldValidationState` particularly when this one is not the one we'd want users to use